### ISSUE TITLE
Jetpack > Dashboard > My Plan: JS errors due to nested a tags

### ISSUE THREAD

**Issue by jeherve** [Automattic/jetpack#33814]
### Steps to reproduce

1. Open your browser's JavaScript console.
2. Go to Jetpack > Dashboard > My Plan
3. Notice the errors:

`Warning: validateDOMNesting(...): <a> cannot appear as a descendant of <a>.`

This is because in `projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js`, we use the `ExternalLink` component inside the `Button` component.

This should be fixed:

- Either by updating the `Button` component itself to support defining an `external` prop that would display an External icon and set the link to open in a new tab. **I think that should be the preferred option since it would then be usable in multiple other areas of the dashboard**.
- Either by moving away from `ExternalLink` in that file, and instead follow a pattern similar to this one, with a manually added `Gridicon`:
https://github.com/Automattic/jetpack/blob/9108e0f9632ab11c7aa03f79f27062ceb9bf4b34/projects/plugins/jetpack/_inc/client/at-a-glance/security-bundle/index.jsx#L72-L80

**Comment by darssen** [comment]
From what I [gathered](https://github.com/Automattic/jetpack/blob/trunk/projects/plugins/jetpack/_inc/client/components/button/index.jsx#L32), the button component does create an anchor link (by creating an `<a>` tag) if passed an `href`, but it treats it as a button if `href` is not passed.
Since `ExternalLink` can be passed an `href` as well, shouldn't something like below work?
```js
<Button onClick={ this.trackAllPurchasesClick } compact rna>
  <ExternalLink href={ getRedirectUrl( 'calypso-purchases' ) }>
	  { __( 'View all purchases', 'jetpack' ) }
  </ExternalLink>
</Button>
```
I can prepare a PR if this approach is good for the scenario in  `projects/plugins/jetpack/_inc/client/my-plan/my-plan-header/index.js ` and some others in `projects/plugins/jetpack/_inc/client/my-plan/my-plan-body.js`
